modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/blueprint/search/request/body/contributor/IndexSXPSearchRequestBodyContributor.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ *
+ *
+ *
+ */
+
+package com.liferay.search.experiences.internal.blueprint.search.request.body.contributor;
+
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.search.SearchContext;
+import com.liferay.portal.kernel.util.ArrayUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.search.index.IndexInformation;
+import com.liferay.portal.search.searcher.SearchRequestBuilder;
+import com.liferay.search.experiences.blueprint.exception.InvalidIndexNameException;
+import com.liferay.search.experiences.internal.blueprint.parameter.SXPParameterData;
+import com.liferay.search.experiences.rest.dto.v1_0.Configuration;
+import com.liferay.search.experiences.rest.dto.v1_0.IndexConfiguration;
+
+import java.beans.ExceptionListener;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * @author Wade Cao
+ */
+public class IndexSXPSearchRequestBodyContributor
+	implements SXPSearchRequestBodyContributor {
+
+	public IndexSXPSearchRequestBodyContributor(
+		IndexInformation indexInformation) {
+
+		_indexInformation = indexInformation;
+	}
+
+	@Override
+	public void contribute(
+		Configuration configuration, SearchRequestBuilder searchRequestBuilder,
+		SXPParameterData sxpParameterData) {
+
+		IndexConfiguration indexConfiguration =
+			configuration.getIndexConfiguration();
+
+		if ((indexConfiguration == null) ||
+			Validator.isBlank(indexConfiguration.getIndexName())) {
+
+			return;
+		}
+
+		RuntimeException runtimeException = new RuntimeException();
+
+		_processIndexName(
+			runtimeException::addSuppressed, indexConfiguration.getExternal(),
+			indexConfiguration.getIndexName(), searchRequestBuilder,
+			searchRequestBuilder.withSearchContextGet(
+				searchContext -> searchContext));
+
+		if (ArrayUtil.isNotEmpty(runtimeException.getSuppressed())) {
+			throw runtimeException;
+		}
+	}
+
+	@Override
+	public String getName() {
+		return "indexConfiguration";
+	}
+
+	private void _processIndexName(
+		ExceptionListener exceptionListener, Boolean external, String indexName,
+		SearchRequestBuilder searchRequestBuilder,
+		SearchContext searchContext) {
+
+		String fullIndexName = null;
+
+		if ((external != null) && external.booleanValue()) {
+			fullIndexName = "external-" + indexName;
```

Comment:
I don't think we need this prefix here as the IndexConfiguration already has the "external" flag. This code here would force the actual index names to begin with "external-" which would maybe not be desired.